### DISCUSSION TITLE
2024-08-19 월요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#288] [기록보드]
2024-08-19 금요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 내용

### 트러블 슈팅

- 내용

### 내일 계획

- 내용
```

```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by Captain-Kim** [comment]
최종주차 To Do List (김병준)

- [ ] 온도지수 계산 로직 구현
- [ ] 글쓰기 퍼널 패턴 뒤로가기 시 이전 스텝으로 가게 수정
- [ ] UT 결과보고서 PPT 제작
- [ ] 레디우스 값, 마진 값 손 보기
- [ ] 리치텍스트 에디터 구현
- [ ] 버디 프로필 페이지 chips 간격 4px
- [ ] 추천인기 버디즈 페이지 버디 선정 기준 나타내기 i 아이콘 호버 등
- [ ] 글쓰기 나이설정에서 input에 0을 못 지우는 문제 해결
- [x] 만든 여정 아코디언 -> 페이지로 다시 제작
- [x] 프로필 사진 변경 -> 프로필 사진 적용으로 바꾸고 버튼으로 바꾸기
- [x] bugfix : 나의 여정 상세에서 팔로우 버튼 스켈레톤이 계속 렌더링 되는 버그 수정
- [x] 팔로워 리스트 렌더링 페이지 구현
- [x] bugfix : 내가 나를 팔로우 가능한 버그 수정 -> 라우트에서 막았음에도 왜? -> GET 요청에 넣었었음. POST로 옮김.
- [x] 추천인기버디즈 하트로 follow 버튼 추가
- [x] 여정 나가기 기능 구현
- [x] 글쓰기 아이콘 여정 리스트로 옮기기 (아이콘, 색상까지 바꾸기)
- [x] 버디 프로필 수정 페이지 만들기 -> 각 내용 클릭 시 온보딩 페이지 컴포넌트 재사용 했다가 다시 돌아와서 수정 페이지에서 ‘완료’버튼 눌러야 수정되는 flow
- [x] 버디 프로필 컴포넌트 블러 (로그인 세션 없으면)
- [x] 버디 프로필 모바일 헤더 X 아이콘 추가
- [x] 버디 프로필 글씨 크기 바꾸기
- [x] 온도지수 아이콘 넣기
- [x] 글쓰기 펜딩 페이지 이미지 디자이너님께 보내드리기
- [x] 여정 삭제하기 기능 구현 -> 구현 완료 -> 클라이언트에서 로직 수행하라는 팀원 피드백 있어 수정까지 완료
- [x] 글쓰기 나이 설정 레인지 좁히기 / 미니멈 20세로 유효성까지 막고 맥시범은 알아서 낮춰보고 유효성X -> 20~70세로 레인지 범위 수정, 유효성은 20~150세로.
- [x] ‘참여하기’ 클릭 시 인원이 가득찼습니다의 경고가 error -> caution으로 수정 -> TripCard에서 에러 메시지를 라우트에서 직접 받아서 showAlert으로 전달하도록 코드가 작성되어 있는데, caution을 전달할 방법이 생각이 안 나 일단 패스.
- [x] 추천인기버디즈 나이까지 표기하기 -> 온보딩 미등록한 버디가 있어 NaN이 출력되는 경우(반환값이 string이 아닌 경우 렌더링 하지 않도록 수정)
- [x] 버디프로필에서 ‘선호하는 버디를 등록해주세요’ 문구를 ‘~ 없습니다’로 변경
- [x] 온도 지수 -> 버디즈 지수로 바꾸기
- [x] 글쓰기 여정 시작일이 오늘일 때 ‘시작 날짜가 현재 날짜보다 이전입니다’ 경고 출력 문제 해결 -> 사용자에게 착오가 있었던 듯. startDate < now로 조건문이 되어 있어 정상 작동함. -> 아님. 문제가 재현되어서 다시 살펴보니 setStartDateTimestamp(value.start.toString());  이렇게만 담고 있어서 생긴 문제였음. 이렇게 하면 시간 데이터가 담기지 않게 되고 ‘2024-08-12’ 이런 형태로만 담기는데 시간 정보를 포함하지 못해서 이런 경우에는 00:00:00초인 자정을 기준으로 날짜가 생성됨. -> 그렇다 보니 자정에서 1초라도 지나면 과거로 보게 되어 이런 현상이 생김. 따라서 시간은 따로 입력받지 않으니 임의로 23:59:59로 설정하여 어떠한 경우에도 이런 에러가 안 뜨도록 수정함.

**Comment by eunohhh** [comment]
## 개인 기록보드 서식 (오은)

### 처리 할 사항(배포버전 체크)
- [x] : 소셜로그인시 인가 상태 안바뀜
- [x] : vercel analytics 적용
- [x] : 로그아웃시에도 알림 숫자 떠있음
- [x] : ai 그림 생성시 타임아웃
- [x] : 스토리 쓰기 시 필터 목록 이상함
- [x] : 스토리 없어도 작성할 수 있게 메인에 빈 카드 띄우기
- [x] : 스토리 생성 완료시 잘못된 곳으로 감
- [x] : 플필페이지 팔로잉/ 팔로우 초기값
- [x] : 비번 변경이 소셜로그인이면 안되게끔
- [x] : 팔로우 / 팔로잉 디자인
- [x] : 여정 수정페이지 전체적으로 검토
- [x] : 여정 수정페이지 한번 진입하면 헤더에 수정이 자꾸 따라옴
- [x] : 마이페이지에서 헤더 톱니바퀴 눌렀을 때 온보딩으로 감 - 이부분은 유저가 톱니바퀴를 클릭했을때 전체 수정할 수 있기를 기대할 것 같아서 유지합니다
- [ ] : 컨트랙트 추가시 알림 동작 이상하게 작동하는 부분 확인
- [x] : 여정 삭제시 에러 관련
- [ ] : 검색 동작.....전체 수정.... 진행중 .....
- [ ] : 디자이너님 수정 사항 - 검색 텍스트 14pt
- [ ] : 디자이너님 수정 사항 - 추천인기버디즈, 여정목록에서 새도우 약하게
- [ ] : 디자이너님 수정 사항 - 여정 상세 모바일에서 스트로크로
- [ ] : 목데이터 - 버디즈
- [ ] : 목데이터 - 여정

### 처리 된 사항
- [x] : 알림 및 메시지 카운트 정상작동 확인
- [x] : 나가기 / 대기중 / 참여하기 동작 제대로 이루어지도록 수정
- [x] : isRead 상태에 따라서 보여지기 토글
- [x] : UI 정돈 - 여정카드, 디테일 모바일/데스크탑
- [x] : 이메일 회원가입 비밀번호 강제하기
- [x] : 탑배너 프리페치쿼리
- [x] : 스토리 관련 로직 전체 검토
- [x] : 튜토리얼 반응형 으로 수정
- [x] : 알림 기능 기본
- [x] : 알림 기능 - 참여자가 여러명일때 순차적으로 띄우기 처리
- [x] : 알림 기능 - 참여자 알림 로직 오류 없게 수정
- [x] : 오늘날짜를 여정 시작일로 할 경우 뜨는 에러 처리 - 에러가 안남 / 재현불가
- [x] : UI 정돈 전체적으로- 메인 모바일/데스크탑
- [x] : 로그아웃 해도 팔로우버튼 에러 (나가져도 팔로우 체크하는듯)
- [x] : 온도지수 계산 sql
- [x] : 스테일 타임 설정
- [x] : '실명으로 작성해 주세요' 표기
- [x] : 팔로우 버튼 스타일 일관되게 수정

**Comment by slsl2** [comment]
### Final To Do List

- [x]  : unread messages 계산해서 렌더링
- [x]  : unread messages 실시간 렌더링
- [x]  : 채팅창 열면 contract의 마지막 읽은 메시지 값 변경
- [x]  : 헤더에도 탭메뉴와 동일하게 표시
- [x]  : 알림 리스트 페이지 모바일
- [x]  : 알림 리스트 페이지 데스크탑
- [x]  : 새 알림 표시
- [x]  : 알림 리스트 페이지에서 읽으면 unread true 처리
- [x]  : unread 변경 시 UI 실시간 반영 <- 은님이 해결해주심
- [x]  : 채팅 페이지 데스크탑
- [ ]  : ㅇㅇ님이 입장하셨습니다
- [x]  : 채팅 입장 시점부터 메시지 뜨도록
- [x]  : 채팅 리스트 최근 메시지 순서대로 정렬
- [x]  : input UI 손보기
- [x]  : n/4 데이터 연결
- [x]  : 마지막 메시지 시간 utils 사용해 변경
- [x]  : 헤더, 탭메뉴 UI 최종 디자인에 맞춰 수정
- [x]  : 자잘한 개선 사항 반영 (interface → type으로 변경 등)

### 오늘 추가
- [x]  : chat/[id] URL 입력 시 권한 없어도 페이지 뜨는 점 개선
- [x]  : home 페이지 서치바에 날짜 디폴트 값을 오늘 날짜로 변경
- [ ]  : 채팅 메시지 보낼 때 내가 보낸 메시지도 알림 숫자가 뜨는 부분 개선 <- 내일 오전까지 완성 예정
